main.rb

```diff
@@ -0,0 +1,98 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    size = self.size
+    while i < size
+      is_a?(Range) ? yield(min + i) : yield(self[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    idx = 0
+    my_each do |elem|
+      yield(elem, idx)
+      idx += 1
+    end
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    if is_a?(Array)
+      selected = []
+      my_each { |ele| selected << ele if yield ele }
+    else
+      selected = {}
+      my_each { |key, value| selected[key] = value if yield(key, value) }
+    end
+    selected
+  end
+
+  # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+
+  def my_all?(arg = nil, &prc)
+    return true if !block_given? && arg.nil? && include?(nil) == false && include?(false) == false
+    return false unless block_given? || !arg.nil?
+
+    if block_given?
+      my_each { |elem| return false if prc.call(elem) == false }
+    elsif arg.class == Regexp
+      my_each { |elem| return false if arg.match(elem).nil? }
+    elsif arg.class <= Numeric || arg.class <= String
+      my_each { |elem| return false if elem != arg }
+    else
+      my_each { |elem| return false if (elem.is_a? arg) == false }
+    end
+    true
+  end
+
+  def my_any?(arg = nil, &prc)
+    return true if !block_given? && arg.nil? && my_each { |elem| return true if elem == true } && empty? == false
+    return false unless block_given? || !arg.nil?
+
+    my_each do |ele|
+      return true if prc.call(ele) == true
+    end
+    false
+  end
+
+  def my_none?(arg = nil, &prc)
+    !my_any?(arg, &prc)
+  end
+
+  def my_count
+    count = 0
+    my_each do |_|
+      count += 1
+    end
+    count
+  end
+
+  def my_map(&prc)
+    return enum_for(:map) unless block_given?
+
+    mapped = []
+    my_each { |ele| mapped << prc.call(ele) }
+    mapped
+  end
+
+  def my_inject
+    acc = shift
+    item_removed = acc
+    my_each { |num| acc = yield(acc, num) }
+    unshift(item_removed)
+    acc
+  end
+end
```

Comment:
#### my_inject

- [ ] combines all elements of enum by applying a binary operation, specified by a block

```ruby
[1, 2, 3].my_inject(1) { |memo, num| memo + num } => 7
```

- Yours is not expecting arguments so it's giving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/77499133-e34c8480-6e16-11ea-940d-6b411df8d2e4.png)

- Real my_inject
![image](https://user-images.githubusercontent.com/52209067/77499178-00815300-6e17-11ea-8647-219d7ae6b2f4.png)

- [ ] when a symbol is specified combines each element of the collection by applying the symbol as a named method

```ruby
[1, 2, 3].my_inject(:+)  => 6
[1, 2, 3].my_inject(1, :+) => 7
```
- Yours is not expecting arguments so it's giving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/77499308-5ce47280-6e17-11ea-8b72-e9cb15b35ea1.png)

- Real my_inject
![image](https://user-images.githubusercontent.com/52209067/77499346-78e81400-6e17-11ea-959a-3409720087b8.png)